hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/AbstractClientCacheProxyBase.java

```diff
@@ -87,9 +113,28 @@ public void close() {
         if (!isClosed.compareAndSet(false, true)) {
             return;
         }
+        waitOnGoingLoadAllCallsToFinish();
         closeListeners();
     }
 
+    private void waitOnGoingLoadAllCallsToFinish() {
+        Iterator<Map.Entry<Future, CompletionListener>> iterator = loadAllCalls.entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<Future, CompletionListener> entry = iterator.next();
+            Future f = entry.getKey();
+            CompletionListener completionListener = entry.getValue();
+            try {
+                f.get(TIMEOUT, TimeUnit.SECONDS);
+            } catch (Throwable t) {
+                logger.finest("Error occurred at loadAll operation execution while waiting it to finish on cache close!", t);
+                if (completionListener != NULL_COMPLETION_LISTENER) {
```

Comment:
sometimes completionListener check is compared to null and sometimes NULL_COMPLETION_LISTENER. I think it would be nice if would be one or the other. See also the onResponse / onFailure